File: server/Domain/Token.cs
using System;

namespace plantio.Domain {
    public class Token {
        private static readonly int elapsedHours = 4;

        public string Value { get; }
        public User Owner { get; }
        public DateTime Expiration { get; }

        public Token(): this("", new User(), DateTime.Now) {
        }

        private Token(string value, User owner, DateTime time) {
            this.Value = value;
            this.Owner = owner;
            this.Expiration = time;
        }

        public static Token Create(string value, User owner) {
            var expiration = DateTime.Now.AddHours(elapsedHours);
            return new Token(value, owner, expiration);
        }
    }
}

File: plantio/src/main/java/com/deltegui/plantio/users/domain/Token.java
package com.deltegui.plantio.users.domain;

import com.fasterxml.jackson.annotation.JsonIgnore;

import java.time.LocalDateTime;

public class Token {
    private final String value;
    private final LocalDateTime expiration;
    private final String user;

    public Token(String value, LocalDateTime expiration, String user) {
        this.value = value;
        this.expiration = expiration;
        this.user = user;
    }

    public String getValue() {
        return value;
    }

    public LocalDateTime getExpiration() {
        return expiration;
    }

    @JsonIgnore
    public String getUser() {
        return user;
    }
}
